packages/l10n/src/create-l10n-injector/create-l10n-injector.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import { getDisplayName } from '../utils';
+
+type InjectorOptions<T> = {
+  displayName: string;
+  propKey: string;
+  propLoadingKey: string;
+  loadLocale: (locale: string, cb: (error?: Error, data?: T) => void) => void;
+};
+
+export default function createL10NInjector<T>({
+  displayName,
+  propKey,
+  propLoadingKey,
+  loadLocale,
+}: InjectorOptions<T>) {
+  return function createHOC<U>(mapPropsToLocale: (props: any) => string) {
+    return (
+      WrappedComponent: React.ComponentType<U>
+    ): React.ComponentClass<U> => {
```

Comment:
I think this return type does not need an explicit annotation

```suggestion
    ) => {
```

Reply:
<img width="734" alt="Screen Shot 2019-06-12 at 8 37 38 AM" src="https://user-images.githubusercontent.com/2425013/59328703-5b651600-8ced-11e9-985f-10a84b5447d0.png">
compiler says no 😢

Reply:
Can you try returning the class directly?

```js
return (WrappedComponent: React.ComponentType<U>) =>
  class L10NComponent extends React.Component<U> {}
```

...instead of `return L10NComponent`